src/handler/sign_psbt.c

```diff
@@ -1383,58 +1383,70 @@ static bool __attribute__((noinline)) display_transaction(
-                               output_description)) {
-                PRINTF("Invalid or unsupported script for external output\n");
-                SEND_SW(dc, SW_NOT_SUPPORTED);
-                return false;
+            for (unsigned int i = 0; i < st->n_external_outputs; i++) {
+                char output_description[MAX_OUTPUT_SCRIPT_DESC_SIZE];
+                /* It is possible to return the following error in the middle of
+                 * already shown screens of previous outputs
+                 */
+                if (!format_script(st->outputs.output_scripts[i],
+                                   st->outputs.output_script_lengths[i],
+                                   output_description)) {
+                    PRINTF("Invalid or unsupported script for external output\n");
+                    SEND_SW(dc, SW_NOT_SUPPORTED);
+                    return false;
+                }
+
+                ui_validate_transaction_simplified_add(
+                    COIN_COINID_SHORT,
+                    is_self_transfer ? 0 : st->outputs.output_amounts[i],
+                    is_self_transfer ? NULL : output_description);
             }
+        } else {
+            ui_validate_transaction_simplified_add(COIN_COINID_SHORT, 0, NULL);
         }
 
-        /** TRANSACTION CONFIRMATION
-         *
-         *  Show transaction amount, destination and fees, ask for final confirmation
-         */
-        if (!ui_validate_transaction_simplified(
-                dc,
-                COIN_COINID_SHORT,
-                st->is_wallet_default ? NULL : st->wallet_header.name,
-                is_self_transfer ? 0 : st->outputs.output_amounts[0],
-                is_self_transfer ? NULL : output_description,
-                st->warnings,
-                fee)) {
+        /* Start the review */
+        if (!ui_validate_transaction_simplified_start(COIN_COINID_SHORT, dc, fee)) {
             SEND_SW(dc, SW_DENY);
             return false;
         }
     } else {
         // Transactions with more than one external output; show one output per page,
         // using the streaming NBGL API.
 
-        // On NBGL devices, show the pre-approval screen
-        // "Review transaction to send Bitcoin"
-        if (!ui_transaction_prompt(dc)) {
-            SEND_SW(dc, SW_DENY);
-            return false;
-        }
-        // If it's not a default wallet policy, ask the user for confirmation, and abort if they
-        // deny
-        if (!st->is_wallet_default && !ui_authorize_wallet_spend(dc, st->wallet_header.name)) {
+        // If it's not a default wallet policy, let's save this info to ask the user for
+        // confirmation
+        if (!st->is_wallet_default && !ui_authorize_wallet_spend(st->wallet_header.name)) {
             SEND_SW(dc, SW_DENY);
             return false;
         }
 
-        if (!display_warnings(dc, st)) {
+        // On NBGL devices, show the pre-approval screen
```

Comment:
By now, all devices are NBGL devices. So this should either be removed, or updated to reflect the correct conditional